packages/fxa-settings/src/components/Settings/FlowRecoveryKeyConfirmPwd/index.tsx

```diff
@@ -62,10 +69,10 @@ export const FlowRecoveryKeyConfirmPwd = ({
     const password = getValues('password');
     logViewEvent(`flow.${viewName}`, 'confirm-password.submit');
     try {
-      if (action === RecoveryKeyAction.Change && account.recoveryKey) {
-        account.deleteRecoveryKey();
-      }
-      const recoveryKey = await account.createRecoveryKey(password);
+      const recoveryKey =
+        actionType === RecoveryKeyAction.Create
+          ? await account.createRecoveryKey(password, false)
+          : await account.createRecoveryKey(password, true);
```

Comment:
Not sure if these are the same or one is any better/clearer than the other. 🤷 
(or even a temp variable for `const isCreateActionType = actionType !== RecoveryKeyAction.Create;` instead of inlining that.)

```suggestion
      const recoveryKey = await account.createRecoveryKey(password, actionType !== RecoveryKeyAction.Create);
```

Reply:
Agree this can be cleaner! Switching it to use a temp variable :)

```
const replaceKey = actionType === RecoveryKeyAction.Change;
const recoveryKey = await account.createRecoveryKey(password, replaceKey);
```